src/python/pants/backend/python/typecheck/pyright/rules_integration_test.py

```diff
@@ -426,3 +431,54 @@ def assert_partition(
         "3.9",
         "b",
     )
+
+@skip_unless_python39_present
+def test_requirements_venv_creation_partition(rule_runner: PythonRuleRunner) -> None:
+    files = {
+        **create_folder("resolveA_py39", "a", "3.9"),
+        **create_folder("resolveB_1", "b", "3.9"),
+        **create_folder("resolveB_2", "b", "3.9"),
+    }
+    rule_runner.write_files(files)
+
+    with temporary_dir() as cache_dir:
+        rule_runner.set_options(
+            [
+                "--python-resolves={'a': '', 'b': ''}",
+                "--python-enable-resolves",
+                f"--named-caches-dir={cache_dir}",
+            ],
+            env_inherit={"PATH", "PYENV_ROOT", "HOME"},
+        )
+
+        resolve_a_py39_dep = rule_runner.get_target(Address("resolveA_py39", target_name="dep"))
+        resolve_a_py39_root = rule_runner.get_target(Address("resolveA_py39", target_name="root"))
+        resolve_b_dep1 = rule_runner.get_target(Address("resolveB_1", target_name="dep"))
+        resolve_b_root1 = rule_runner.get_target(Address("resolveB_1", target_name="root"))
+        resolve_b_dep2 = rule_runner.get_target(Address("resolveB_2", target_name="dep"))
+        resolve_b_root2 = rule_runner.get_target(Address("resolveB_2", target_name="root"))
+        request = PyrightRequest(
+            PyrightFieldSet.create(t)
+            for t in (
+                resolve_a_py39_root,
+                resolve_b_root1,
+                resolve_b_root2,
+            )
+        )
+
+        def run_partition_test():
```

Comment:
I don't think this is testing the right thing. This is just generating the pyright input partitions and checking that they contain what you'd expect. But you want to test that actually running Pyright works under the relevant conditions. 

Look at the tests that call `run_pyright()` for guidance.

Reply:
Good point. I changed the test to actually run Pyright and clear the cache in between, but I think the examples might be a bit too simple and I'm not sure how to force the test to break without introducing some third-party imports.